### DISCUSSION TITLE
Ignoring a variable number of columns at end of the CSV?

### DISCUSSION THREAD

**Discussion by thedavegray** [JoshClose/CsvHelper#2109] [General]
Hello

Is there a way of ignoring one or more columns from the end of a CSV? 

I have a file that comes from an external source that occasionally has a one or more extra columns (junk values) at the end.   I added this to the end of my class definition but I still get an error about bad data and that it can be ignore. 

<img width="271" alt="gh1" src="https://user-images.githubusercontent.com/4395873/211351523-ba1b1162-0d4c-4e70-a4a0-c84ce283e877.png">
<img width="514" alt="gh2" src="https://user-images.githubusercontent.com/4395873/211351651-097b83e9-6a68-41bb-ab11-0e0230e3fa71.png">

**Comment by JoshClose** [comment]
Bad data happens with parsing. There is a field with an invalid format. You can try to ignore the bad data by settings `BadDataFound = null` like you did, but it's possible that could mess up reading the rest of the file.
